### PR TITLE
Adding documentation website

### PR DESCRIPTION
**Pull request by naskio**
- Adding website at `docs/`.
- Updating `README.md`.
- Moving `logo/` folder and other markdown files into the website.
- Updating links to point to the new domain `consuldot.net`.
- Adding GitHub workflow to auto-deploy the website into Pages (The version being used in the website is extracted automatically from the latest tag).

# TODO
In order to deploy the website to GitHub Pages, we need to perform the following tasks:

- [ ] Merge this PR.
- [x] Configure DNS for the new domain `consuldot.net`.
- [x] Enable GitHub Pages.

### CODE REVIEW

**Comment by jgiannuzzi** [comment]
This looks awesome, thank you so much for your contribution @naskio! 🤩

@pavlovic-ivan could you please review the workflow?
@mfkl @marcin-krystianc could you please review the content?
@c-rindi could you please setup the domain?

**Comment by jgiannuzzi** [comment]
I have enabled GitHub Pages for the repo.

**Review by marcin-krystianc — APPROVED** [review]
It looks great, thanks!

**Review by mfkl — DISMISSED** [review]

**Comment by c-rindi** [comment]
I have the set up the DNS - CNAME record. 
Thanks for good work @naskio and the clear instructions @jgiannuzzi. 😄

**Comment by jgiannuzzi** [comment]
@c-rindi I missed one instruction 😓 — could you please add a similar `CNAME` for the `www` subdomain?

File: .github/workflows/deploy-website.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+name: Deploy website to Pages
+
+on:
+  push: # Runs on pushes targeting the default branch
+    branches: [ $default-branch ]
```
Comment (pavlovic-ivan) [inline .github/workflows/deploy-website.yml:5]: @naskio the `$default-branch` is not supported currently in Github, you have to list the name of the default branch explicitly.

**Review by pavlovic-ivan — CHANGES_REQUESTED** [review]
@naskio thank you for the PR. one small change required before i approve. the rest looks good to me!

**Review by pavlovic-ivan — APPROVED** [review]
@naskio looks good. thanks 🙏

**Comment by jgiannuzzi** [comment]
Thanks @c-rindi, the DNS checks are now successful!

Thank you @marcin-krystianc, @mfkl, and @pavlovic-ivan for your reviews, and most importantly thank you @naskio for this amazing contribution!

**Comment by jgiannuzzi** [comment]
Great point @naskio — done!

**Pull request merged**